### PR TITLE
add OCTGN guids to card data

### PR DESCRIPTION
**Pull request by brine**
as discussed in #27, this adds the existing OCTGN guids to card data.

### CODE REVIEW

**Comment by Alsciende** [comment]
I reverted it. As discussed in https://github.com/Alsciende/fiveringsdb-data/issues/27, the decks don't know which packs the cards come from. So putting the octgn_id in PackCard will not help in exporting those values with a deck content.

**Comment by brine** [comment]
Thats fine, I can get them from the card api

**Comment by brine** [comment]
I will be able to fetch the API card data with the deck generator plugin I'll need to develop, so I can pull the GUIDs from there.  There's no need to export them with the deck content.

So that considered, it makes more practical sense to have the GUIDs fall under PackCard as each card in OCTGN is uniquely linked to a specific card data and image.

A side note, I understand that the deck content itself doesn't retain info for specific card printings.  When that deck is viewed on the website, however, hovering over a card displays its pack name and position -- data that would have had to come from PackCard.  Does it just display the card's original/default printing, and if so, why could the GUID not come from there?

**Comment by Alsciende** [comment]
Starting from the end, the app does indeed try to guess the main pack of a card. I don't even remember how it does it. Most cards have one only pack, so it's often easy to fall right ;) 

The GUID cannot from here because the server is not going to do that kind of computation/guessing game when it exports a deck. It really doesn't make sense for the server or the app. It would only make sense for a third-party tool that needs to see cards as elements of a pack. 

I can include the GUID in the PackCard entities, and make them available in the https://api.fiveringsdb.com/pack-cards and https://api.fiveringsdb.com/cards endpoints. But then it will fall on you to decide which pack do you want to take the GUID of for each card of a deck. If there ever is a Core Set 2.0, that task will become much more difficult.

I guess you can also decide that the promos cards don't have a GUID. And, by extension, that cards have a GUID for only one pack. And so, from there, the GUID should be on the card itself, not the packCard. And if so, then all your problems are solved. Each deck contains cards, each card has a GUID.

**Comment by brine** [comment]
Another side note: What's the plan for down the road for if the game designers decide to print different versions of a character with the same name, like they do in AGOT?  The id matching method used in this API seems to rely on the card name being unique.

I've got a bunch of stuff I've been trying to figure out how to explain, when I get home tonight I'll try to formulate it into something coherent.  To summarize it, I don't think there's any practical benefit to putting them in the Card entity, as it's not compatible with how octgn organizes its cards.  It's much closer to how the thronesdb card data is organized.

**Comment by Alsciende** [comment]
The plan is to have an id that is not exactly derived from the name, for those cards. I'll add a property suffix and the id will become derived from name + suffix.

**Comment by brine** [comment]
I've been working on a library which reads the json data off the API and combines it with OCTGN's card database.  I originally designed it for the AGOT octgn plugin, where the `octgn_id` field is essentially mandatory for the deck exporter feature to function properly.  This field is also the easiest way to bridge the two card formats together, as it's (in theory) guaranteed that a GUID value from the API, queried to OCTGN's card database, will return exactly 1 OCTGN card entity.

When I adapted the library to interact with the fiveringsDB API, obviously the GUID querying wouldn't work as there's no GUIDs to begin with.  So as a fallback, if the GUID is missing, I use the card's pack and position when I query OCTGN's database instead.  So far it's been working fine, but I'm not nearly as confident in its accuracy than I would be with direct GUID queries.

The proposed deck converter app would still require card GUIDs, since they're mandatory in OCTGN's deck file format.  We've established that the GUIDs cannot be included with the exported fiveringsDB deck data, so that option is off the table.  I should be able to use the library to fill in the missing blanks.  Since I'm handling that missing GUID fallback, I guess you technically don't even need to have the GUIDs included in the database.

TL;DR: I don't think we need to add the GUIDs to the database after all.
